### PR TITLE
(chore) cost surface data migration script

### PR DESCRIPTION
**Pull request by alexeh**
Script for handling the data migration for existing projects and scenarios.

NOTE: Since this will be run most likely from NOT the host, I haven't add any requirements.txt. 

### Overview

**API:**
For each scenario, create a cost surface within the project, and set it as the scenarios cost surface. We use the scenario name as cost surface name

For each project, create a new cost surface using 'Default' as name, and set it as default cost surface for the project.

**GEO & API**
for each scenario in api, get its cost surface and compute min and max from scenario_pu_cost_data.
update api.cost_surface with said min,max

for each project, get its default cost surface and insert it's default cost in cost_surface_pu_data
update the the api.cost_surface computing the min and max from the new cost

**Error handling**
There might be scenarios in API that has no cost in GEO. **I have assumed** this should not happen, so I am deleting the scenario and its cost_surface in API

I don't know if the same could happen for projects 

### Testing instructions

Basically the above should happen

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: data/scripts/cost_surface_data_migration.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            SELECT MIN(spcd.cost) as min, MAX(spcd.cost) as max
+            FROM scenarios_pu_data spd
+            LEFT JOIN scenarios_pu_cost_data spcd on spd.id = spcd.scenarios_pu_data_id
+            WHERE spd.scenario_id = '{scenario_id}';
+        """)
+        min_max = cur_geo_db.fetchone()
+        min_val, max_val = min_max if min_max else (None, None)
+
+        # If min and max are null, remove scenario and cost_surface from api_db
+        if min_val is None or max_val is None:
```
Comment (alexeh) [inline data/scripts/cost_surface_data_migration.py:113]: Please give extra love to this part

File: data/scripts/cost_surface_data_migration.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+        # Insert data into cost_surface_pu_data
+        cur_geo_db.execute(f"""
+                   INSERT INTO cost_surface_pu_data (puid, cost, cost_surface_id)
+                   SELECT ppu.id, round(pug.area / 1000000) as area, '{cost_surface_id}'
+                   FROM projects_pu ppu
+                   INNER JOIN planning_units_geom pug ON pug.id = ppu.geom_id
+                   WHERE ppu.project_id = '{project_id}';
+               """)
+
```
Comment (alexeh) [inline data/scripts/cost_surface_data_migration.py:143]: TODO: insert for each scenario in apiDB

File: data/scripts/cost_surface_data_migration.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    alter_table_sql2 = '''ALTER TABLE scenarios
+        ALTER COLUMN cost_surface_id SET NOT NULL, ADD CONSTRAINT fk_cost_surface FOREIGN KEY (cost_surface_id) REFERENCES cost_surfaces(id)'''
+    cur_api_db.execute(alter_table_sql2)
+    cur_api_db.execute("COMMIT;")
+    print("Successfully migrated data from marxan-api for:")
+    print(len(all_scenarios), "scenarios")
+    print(len(all_projects), "projects")
+except Exception as e:
+    print(e)
+    cur_api_db.execute("ROLLBACK;")
```
Comment (alexeh) [inline data/scripts/cost_surface_data_migration.py:85]: TODO: All ops in one single transaction: one commit, one rollback

**Comment by hotzevzl** [comment]
closing as we included this, with some tweaks, in #1494
